app/controllers/support_interface/sessions_controller.rb

```diff
@@ -10,9 +10,14 @@ def new
     end
 
     def destroy
-      DfESignInUser.end_session!(session)
+      post_signout_redirect = if dfe_sign_in_user&.needs_dsi_signout?
```

Comment:
How could we get here and not have a `dfe_sign_in_user`?

Reply:
Well, dfe_sign_in_user is really just `DfESignInUser.load_from_session(session)` so if you visit `/provider/sign-out` or `/support/sign-out` while unauthenticated (possible because we skip auth in this controller) this will be `nil` and without the paranoia you get a 500.

I encountered this while testing the feature when the test DSI env wasn't properly configured, which stopped the flow of redirects and presented me with a DSI error page. I then clicked the browser's back button, which showed me the applications list, with a Sign out link at the top. So I tried to sign out again.

It's not super important but I'm trying to prevent generating 500 errors from simply visiting sign-out urls.

Reply:
On second thoughts, I'll remove the `&.` and ensure `:authenticate_*_user!` methods are called for session `:destroy` actions.